cfpb/cfpb-chart-builder
# Fix/projected value month

Fixes https://github.com/cfpb/cfpb-chart-builder/issues/57 - bug with the displayed projected value month in the bar + line charts. Improves date functions and unit tests.


## Changes

- Refactor `getProjectedTimestamp()` to reliably get the six months ago date for projected data labeling. Previously this function calculated six months ago by counting the days from the beginning of the month of the latest month of data available. This was prone to causing bugs because dates don't really work that way. Now, the month is found by checking the last six months of data available and grabbing the timestamp directly from the data object.
- Unit tests were updated to reflect this expected timestamp and to check additionally that the expected data value matches the expected timestamp for that month.
- Tweaks some CSS values to accommodate the length September text. This is the longest month name we've had in the UI so it wasn't positioned correctly with our existing CSS :)

## Testing

- check out branch
- run `gulp build` and `gulp watch` to see the changes in the browser - the charts should now say "Values after September 2016 are projected" instead of August (which was wrong!)
- run `gulp test` to see unit tests pass. break one if you want to check it tests what we need it to.

## Review

- @mistergone or @anselmbradford 

[Preview this PR without the whitespace changes](?w=0)

## Screenshots
![screen shot 2017-07-17 at 11 46 13 am](https://user-images.githubusercontent.com/702526/28276861-294bea52-6ae6-11e7-981c-0b4ff93935bc.png)
![screen shot 2017-07-17 at 11 46 07 am](https://user-images.githubusercontent.com/702526/28276862-294e1ebc-6ae6-11e7-8a31-0f474752ef69.png)


## Notes

-

## Todos

-

## Checklist
